### PR TITLE
Dark theme for segmented control - #299

### LINKED ISSUE #299: Support dark theme for Segment Control

### PR DESCRIPTION
**Pull request by BAL4-R**
Added dark theme for segmented control, closes #299 

Screenshot:
Screenshot 2023-12-01 at 2 43 09 PM

### CODE REVIEW

File: Trimble.Modus.Components/Controls/TMSegmentedControl/TMSegmentedItem.xaml.cs [hunk 1]
```diff
@@ -15,13 +15,44 @@ public partial class TMSegmentedItem
...
     public static readonly BindableProperty ColorThemeProperty = BindableProperty.Create(nameof(ColorTheme), typeof(SegmentColorTheme), typeof(TMSegmentedItem), SegmentColorTheme.Primary, propertyChanged: OnColorThemeChanged);
     public static readonly BindableProperty IconProperty = BindableProperty.Create(nameof(Icon), typeof(ImageSource), typeof(TMSegmentedItem), null, propertyChanged: OnSegmentedItemPropertyChanged);
-    internal static readonly BindablePropertyKey CurrentBackgroundColorPropertyKey = BindableProperty.CreateReadOnly(nameof(CurrentBackgroundColor), typeof(Color), typeof(TMSegmentedItem), ResourcesDictionary.ColorsDictionary(ColorsConstants.Transparent));
-    public static readonly BindableProperty CurrentBackgroundColorProperty = CurrentBackgroundColorPropertyKey.BindableProperty;
+    public static readonly BindableProperty CurrentBackgroundColorProperty = BindableProperty.Create(nameof(CurrentBackgroundColor), typeof(Color), typeof(TMSegmentedItem), ResourcesDictionary.ColorsDictionary(ColorsConstants.Transparent),
+        propertyChanged: (bindable, _, newValue) =>
+        {
```
Comment (Sofiya-kumar) [inline Trimble.Modus.Components/Controls/TMSegmentedControl/TMSegmentedItem.xaml.cs:20]: Can you add as a method

File: Trimble.Modus.Components/Styles/DarkTheme.xaml [hunk 2]
```diff
@@ -72,4 +72,16 @@
...
   <Color x:Key="BadgeAlternateTextColor">#252A2E</Color>
   <Color x:Key="BadgeTextColor">#FFFFFF</Color>
+
+  <!--Segmented Control-->
+  <Color x:Key="SegmentSeparatorColor">#6A6E79</Color>
+  <Color x:Key="SegmentItemTextColor">#FFF</Color>
+  <Color x:Key="SegmentItemSelectedTextColor">#FFF</Color>
+  <Color x:Key="SegmentItemSelectedIconTintColor">#FFF</Color>
+  <Color x:Key="SegmentItemIconTintColor">#FFF</Color>
+
```
Comment (Sofiya-kumar) [inline Trimble.Modus.Components/Styles/DarkTheme.xaml:82]: Can you remove these extra spaces that followed in other controls colors as well?

**Review by Sofiya-kumar — CHANGES_REQUESTED** [review]

File: Trimble.Modus.Components/Controls/TMSegmentedControl/TMSegmentedItem.xaml.cs [hunk 3]
```diff
@@ -183,10 +207,17 @@ void UpdateCurrentItemStyle()
     /// </summary>
     void UpdateBackgroundColor()
     {
-        CurrentBackgroundColor = !IsSelected
-            ? ResourcesDictionary.ColorsDictionary(ColorsConstants.Transparent)
-            : SelectedBackgroundColor
-                ?? ResourcesDictionary.ColorsDictionary(ColorsConstants.TrimbleBlue);
+
+        string baseStyle = (ColorTheme == SegmentColorTheme.Primary) ? "Primary" : "Secondary";
```
Comment (rthanga1) [inline Trimble.Modus.Components/Controls/TMSegmentedControl/TMSegmentedItem.xaml.cs:211]: Can you try the VisualStateManager like in the other control for toggling b/w Normal and Selected?

**Review by rthanga1 — CHANGES_REQUESTED** [review]

**Comment by rthanga1** [comment]
Also getting an error in the console while opening the SegmentControl page

`Microsoft.Maui.Controls.BindableObject: Warning: Cannot convert [Color: Red=0, Green=0.3882353, Blue=0.6392157, Alpha=1] to type 'Microsoft.Maui.Controls.Style'
Microsoft.Maui.Controls.BindableObject: Warning: Cannot convert [Color: Red=0, Green=0.3882353, Blue=0.6392157, Alpha=1] to type 'Microsoft.Maui.Controls.Style'
Microsoft.Maui.Controls.BindableObject: Warning: Cannot convert [Color: Red=0, Green=0.3882353, Blue=0.6392157, Alpha=1] to type 'Microsoft.Maui.Controls.Style'
Microsoft.Maui.Controls.BindableObject: Warning: Cannot convert [Color: Red=0, Green=0.3882353, Blue=0.6392157, Alpha=1] to type 'Microsoft.Maui.Controls.Style'
`

File: Trimble.Modus.Components/Controls/TMSegmentedControl/TMSegmentedItem.xaml [hunk 4]
```diff
@@ -1,8 +1,8 @@
-<?xml version="1.0" encoding="utf-8"?>
+﻿<?xml version="1.0" encoding="utf-8"?>
```
Comment (rthanga1) [inline Trimble.Modus.Components/Controls/TMSegmentedControl/TMSegmentedItem.xaml:1]: Getting this error in the console

`Microsoft.Maui.Controls.BindableObject: Warning: Cannot convert [Color: Red=0, Green=0.3882353, Blue=0.6392157, Alpha=1] to type 'Microsoft.Maui.Controls.Style'
Microsoft.Maui.Controls.BindableObject: Warning: Cannot convert [Color: Red=0, Green=0.3882353, Blue=0.6392157, Alpha=1] to type 'Microsoft.Maui.Controls.Style'
Microsoft.Maui.Controls.BindableObject: Warning: Cannot convert [Color: Red=0, Green=0.3882353, Blue=0.6392157, Alpha=1] to type 'Microsoft.Maui.Controls.Style'
Microsoft.Maui.Controls.BindableObject: Warning: Cannot convert [Color: Red=0, Green=0.3882353, Blue=0.6392157, Alpha=1] to type 'Microsoft.Maui.Controls.Style'

`

**Review by rthanga1 — CHANGES_REQUESTED** [review]

File: Trimble.Modus.Components/Controls/TMSegmentedControl/TMSegmentedItem.xaml (hunk 4 above)
Comment (rthanga1) [reply Trimble.Modus.Components/Controls/TMSegmentedControl/TMSegmentedItem.xaml:1]: To check later. @BAL4-R Can you create a tech debt issue for this? We can close this one for now.

**Review by rthanga1 — APPROVED** [review]

**Review by rthanga1 — APPROVED** [review]

**Review by rthanga1 — APPROVED** [review]

**Pull request merged**